pennylane/devices/default_mixed.py

```diff
@@ -116,8 +153,129 @@ def _get_kraus(self, operation):  # pylint: disable=no-self-use
 
         return [operation.matrix]
 
-    # Note: placeholder to be able to run tests
-    # pylint:disable=pointless-statement
-    def apply(self, operations, **kwargs):
+    def _apply_channel(self, kraus, wires):
+        r"""Apply a quantum channel specified by a list of Kraus operators to subsystems of the
+        quantum state. For a unitary gate, there is a single Kraus operator.
+
+        Args:
+            kraus (list[array]): Kraus operators
+            wires (Wires): target wires
+        """
+
+        channel_wires = self.map_wires(wires)
+
+        # Computes K^\dagger, needed for the transformation K \rho K^\dagger
+        kraus_dagger = [self._conj(self._transpose(k)) for k in kraus]
+
+        # Changes tensor shape
+        kraus_shape = [len(kraus)] + [2] * len(channel_wires) * 2
+        kraus = self._cast(self._reshape(kraus, kraus_shape), dtype=self.C_DTYPE)
+        kraus_dagger = self._cast(self._reshape(kraus_dagger, kraus_shape), dtype=self.C_DTYPE)
+
+        # Tensor indices of the state. For each qubit, need an index for rows *and* columns
+        state_indices = ABC[: 2 * self.num_wires]
+
+        # row indices of the quantum state affected by this operation
+        row_wires_list = channel_wires.tolist()
+        row_indices = "".join(ABC_ARRAY[row_wires_list].tolist())
+
+        # column indices are shifted by the number of wires
+        col_wires_list = [w + self.num_wires for w in row_wires_list]
+        col_indices = "".join(ABC_ARRAY[col_wires_list].tolist())
+
+        # indices in einsum must be replaced with new ones
+        new_row_indices = ABC[2 * self.num_wires : 2 * self.num_wires + len(channel_wires)]
```

Comment:
(Minor suggestion): could be worth to factor out a couple of recurring terms for readability, e.g.:
```python
density_dim = 2 * self.num_wires
num_channel_wires = len(channel_wires)
```